pep-0646.rst

```diff
@@ -683,11 +687,21 @@ these type annotations.
 Backwards Compatibility
 =======================
 
-The grammar changes necessary for unpacking of type variable tuples using
-the star operator are almost all covered by PEP 637. The one exception
-is '``*args: *Ts``', which would require an additional change. (This
-change could be avoided by simply mandating this be written
-as ``Unpack[Ts]``, however.)
+In order to use the star operator for unpacking of ``TypeVarTuple`` instances,
+we would need to make two grammar changes:
+
+1. Star expressions must be made valid in index operations,
+   e.g. ``Tuple[*Ts]`` (but *not* in slice operations in general - e.g.
+   ``Tuple[*Ts:*Ts]`` is nonsensical and should remain invalid).
```

Comment:
Also clarify that `Tuple[x, *y, z]` is supported.

we probably need to clarify that this syntax would be accepted anywhere, with semantics as described in PEP 637 (there are subtle edge cases that were reasoned through for PEP 637).

Reply:
What kinds of place do you have in mind when you say "accepted anywhere"? Aren't starred expressions still only valid in the context of indexing in PEP 637?

In terms of "subtle edge cases" - I see there are these examples in PEP 637:
```python
obj[*()]
obj[*(x,)]
obj[*(x,),]
```
Are these the ones you had in mind?

More generally, I'm started to wonder whether it would be better to do a from-scratch implementation of PEP 646 in CPython (rather than basing it on the PEP 637 implementation) ato get a better sense of exactly what would be required. Do you think this would be worth it - would this make a big difference to the SC? Or would it be better to get the SC's early opinion before spending time on implementation?